### PR TITLE
Empty Email check implemented

### PR DESCRIPTION
**Pull request by gagandeepp**
Fixes: https://github.com/gopasspw/gopass/issues/2949

### CODE REVIEW

File: internal/action/setup.go [hunk 1]
```diff
@@ -137,6 +138,10 @@ func (s *Action) initGenerateIdentity(ctx context.Context, crypto backend.Crypto
 		}
 
 		email, err = termio.AskForString(ctx, "📧 What is your email?", email)
+
+		if strings.TrimSpace(email) == "" {
+			return fmt.Errorf("failed to create a usable key pair")
```
Comment (jcrben) [inline internal/action/setup.go:143]: Disclaimer: I'm not a maintainer. But I think this message could be more user-friendly. Something like: "You did not provide an email. An email is required to generate a gpg key pair, so please provide one."

**Comment by gagandeepp** [comment]
@dominikschulz  Please review this

File: internal/action/setup.go (hunk 1 above)
Comment (gagandeepp) [reply internal/action/setup.go]: Done

File: internal/action/setup.go [hunk 2]
```diff
@@ -137,6 +138,10 @@ func (s *Action) initGenerateIdentity(ctx context.Context, crypto backend.Crypto
 		}
 
 		email, err = termio.AskForString(ctx, "📧 What is your email?", email)
+
+		if strings.TrimSpace(email) == "" {
```
Comment (dominikschulz) [inline internal/action/setup.go:142]: nit: I think checking for the email string should happen after error handling in the next block, i.e. please move these lines below line 147.

Comment (gagandeepp) [reply internal/action/setup.go]: Done @dominikschulz

**Comment by jcrben** [comment]
What's the story on the age backend? Does the bug also happen there? I don't really know much about age, not sure if it requires an email.

**Comment by gagandeepp** [comment]
> What's the story on the age backend? Does the bug also happen there? I don't really know much about age, not sure if it requires an email.

@jcrben honestly no idea about it

**Comment by gagandeepp** [comment]
@dominikschulz please review

File: internal/action/setup.go (hunk 1 above)
Comment (gagandeepp) [reply internal/action/setup.go]: @jcrben  Can you please resolve this conversation?

File: internal/action/setup.go (hunk 2 above)
Comment (gagandeepp) [reply internal/action/setup.go]: @dominikschulz  Can you please resolve this conversation?

**Review by dominikschulz — APPROVED** [review]
LGTM

**Comment by AnomalRoil** [comment]
No need for emails with the `age` crypto backend, unless one is using the `gitfs` storage backend with it.

**Comment by dominikschulz** [comment]
@AnomalRoil I don't think we have the infrastructure to plumb through these different backend requirements, yet.

**Pull request merged**